src/Model/Table/SentencesTable.php

```diff
@@ -377,12 +379,56 @@ public function findFilteredTranslations($query, $options) {
                 $indirectTranslations = Hash::sort($indirectTranslations, '{n}.lang', 'asc');
                 
                 $result['translations'] = [$directTranslations, $indirectTranslations];
+                $result['extraTranslationsCount'] = $this->getextraTranslationsCount($result);
+                $result['expandLabel'] = $this->getExpandLabel($result['extraTranslationsCount']);
+                if (CurrentUser::isMember()) {
+                    $result['menu'] = $this->getMenu($result);
+                    $result['lists'] = $this->getLists($result->id);
+                }
 
                 return $result;
             });
         });
     }
 
+    private function getextraTranslationsCount($sentence)
+    {
+        $maxDisplayed = 5;
+        $directTranslationCount = count($sentence->translations[0]);
+        $indirectTranslationCount = count($sentence->translations[1]);
+        return $directTranslationCount + $indirectTranslationCount - $maxDisplayed;
+    }
+
+    private function getExpandLabel($extraTranslationsCount)
+    {
+        return format(__n(
+            'Show 1 more translation',
+            'Show {number} more translations',
+            $extraTranslationsCount,
+            true
```

Comment:
This `true` parameter passed to gettext functions is a relic of a previous version of CakePHP that was used to make the function echo instead of return. You can safely remove it.